src/vos/tests/vts_mvcc.c

```diff
@@ -387,6 +387,48 @@ read_ts_a_f(struct io_test_args *arg, struct tx_helper *txh, char *path,
 			 &iod, iod.iod_nr);
 }
 
+static int
+checkexisto_f(struct io_test_args *arg, struct tx_helper *txh, char *path,
+	      daos_epoch_t epoch)
+{
+	return read_ts_o(arg, txh, path, epoch, VOS_OF_FETCH_CHECK_EXISTENCE |
```

Comment:
you set VOS_OF_FETCH_SET_TS_ONLY here but not in the others.  Is it needed?   I need to check whether setting only VOS_OF_FETCH_CHECK_EXISTENCE results in a timestamp update.  I would expect it should since it's a read.

Reply:
@Nasf-Fan are these flags always used together?

Reply:
FETCH_SET_TS_ONLY is used by the distributed transaction for read-only sub operations, that is nothing related with FETCH_CHECK_EXISTENCE.

FETCH_CHECK_EXISTENCE is used by the distributed transaction to handle conditional modification inside TX.

Reply:
> right, I'm not sure, but if I don't set ts_only when dkey is NULL, I get an "Invalid parameters" error

That should NOT be the reason to use the two flags together.

Currently, the FETCH_CHECK_EXISTENCE is only can be used to check whether the "akey" or "dkey" existence. But If you set "dkey" as NULL, you will get failure. I guess that you hit if you did not set the two flags together.

So I suggest to enhance the logic for FETCH_CHECK_EXISTENCE handling in vos_fetch_begin() to support the case of object existence check.